clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -444,14 +444,7 @@ public static String stackTrace(Throwable e) {
      * @return The unstarted thread
      */
     public static Thread newThread(String name, Runnable runnable, boolean daemon) {
```

Comment:
It's good to make this explicit, so maybe we could have `KafkaThread.daemon` and `KafkaThread.nonDeamon` as static factory methods.